rest/api_test.go

```diff
@@ -5313,3 +5314,139 @@ func TestAttachmentContentType(t *testing.T) {
 		assert.Equal(t, "", contentDisposition)
 	}
 }
+
+func TestDeleteNonExistentDoc(t *testing.T) {
+	rt := NewRestTester(t, nil)
+	defer rt.Close()
+
+	response := rt.SendAdminRequest("DELETE", "/db/fake", "")
+	assertStatus(t, response, http.StatusOK)
+
+	response = rt.SendAdminRequest("GET", "/db/fake", "")
+	assertStatus(t, response, http.StatusNotFound)
+
+	var body map[string]interface{}
+	_, err := rt.GetDatabase().Bucket.Get("fake", &body)
+
+	if base.TestUseXattrs() {
+		assert.Error(t, err)
+		assert.Nil(t, body)
+	} else {
+		assert.NoError(t, err)
+	}
+}
+
+// CBG-1153
+func TestDeleteEmptyBodyDoc(t *testing.T) {
+	rt := NewRestTester(t, nil)
+	defer rt.Close()
+
+	var body db.Body
+	response := rt.SendAdminRequest("PUT", "/db/doc1", "{}")
+	assertStatus(t, response, http.StatusCreated)
+	assert.NoError(t, json.Unmarshal(response.BodyBytes(), &body))
+	rev := body["rev"].(string)
+
+	response = rt.SendAdminRequest("DELETE", "/db/doc1?rev="+rev, "")
+	assertStatus(t, response, http.StatusOK)
+
+	response = rt.SendAdminRequest("GET", "/db/doc1", "")
+	assertStatus(t, response, http.StatusNotFound)
+
+	var doc map[string]interface{}
+	_, err := rt.GetDatabase().Bucket.Get("doc1", &doc)
+
+	if base.TestUseXattrs() {
+		assert.Error(t, err)
+	} else {
+		assert.NoError(t, err)
+	}
+}
+
+func TestPutEmptyDoc(t *testing.T) {
+	rt := NewRestTester(t, nil)
+	defer rt.Close()
+
+	response := rt.SendAdminRequest("PUT", "/db/doc", "{}")
+	assertStatus(t, response, http.StatusCreated)
+
+	response = rt.SendAdminRequest("GET", "/db/doc", "")
+	assertStatus(t, response, http.StatusOK)
```

Comment:
Validating that an update to a doc with body "{}" behaves properly seems like a good idea.